plugins/types.ts

```diff
@@ -14,11 +13,4 @@ export interface ActionPluginOutput {
   output?: { [key: string]: any }
 }
 
-export type Status =
-  | 'Draft'
-  | 'Withdrawn'
-  | 'Submitted'
-  | 'Changes Required'
-  | 'Re-submitted'
-  | 'Completed'
-  | 'Expired'
+export type ActionPluginType = (props: ActionPluginInput) => Promise<ActionPluginOutput>
```

Comment:
What does this do?

Reply:
single type definition for the method (including props and return type), used it in index files to check that plugin conforms to API (since it's dynamically imported it's not checked by TS for overall build, and individual plugin)
Also ended up using it in a few actions (although i didn't need to use it in some)
```
async function action(props: ActionPluginInput): Promise<ActionPluginOutput> { ..
```
vs
```
const action: ActionPluginType = async(props)  => { ...
```